src/api/HttpApi.ts

```diff
@@ -86,17 +86,43 @@ export default abstract class HttpApi {
     return this.loader.load(this.makePath(path, args));
   }
 
+  protected validateConnectionaArgs({ first, after, last, before }: Args) {
+    if (after && before) {
+      throw new TypeError(
+        '`after` and `before` cursors cannot be specified together',
+      );
+    }
+    if (first != null && last != null) {
+      throw new TypeError('`first` and `last` cannot be specified together.');
+    }
```

Comment:
these are not strictly invariants but the Connections spec strongly encourages this constraint